File: Data-Structures/03_Recursion/02_Fibonacci.cpp
#include <iostream>

int fibonacci(int n) {
    // Base Case
    if (n <= 1) {
        return n;
    }

    // Recursive step
    return fibonacci(n-1) + fibonacci(n-2);
}

int main() {
    int result = fibonacci(5);
    printf("5th Fibonacci Number: %d", result);

    return 0;
}

File: DSA-for-Beginners/03_Recursion/02_Fibonacci.py
def fibonacci(n):
    # Base case
    if n <= 1:
        return n
    
    # Recursive step
    return fibonacci(n - 1) + fibonacci(n - 2)

def main():
    n = 12
    result = fibonacci(n)
    print(f"Fibonacci {n} = {result}")

if __name__ == "__main__":
    main()